### ISSUE TITLE
Glitch during the animation when setting a wallpaper and returning to launcher 

### ISSUE THREAD

**Issue by vinayakakv** [8bitPit/Niagara-Issues#2155]
### Describe the bug

When setting the wallpaper from Google Photos app and coming back to the launcher, the launcher animation is accompanied by a glitch.

### Steps to reproduce

1. Go to Google photos app
2. Select a photo
3. Swipe up to reveal more options
4. In the options list, select "Use As"
5. In there choose "Set wallpaper" option provided by Google Photos app
6. Set the wallpaper
7. Immediately go to the home screen to open the launcher 

Note that the glitch does not happen if we do not immediately navigate to the homescreen by spending some time in other screens.

### Does the issue also occur with other 3rd-party launchers?

Problem doesn't occur with other launchers

### Device name

Google Pixel 6a

### Android version

Android 13

### Screenshots / Screen recordings

https://user-images.githubusercontent.com/24838519/208226000-5dff09ee-a45b-432a-a6ff-41b4a3dff7de.mp4

**Comment by Amorphous404** [comment]
Other third party launchers like Lawnchair or Nova don't face this issue?

**Comment by vinayakakv** [comment]
I tested with Nova Launcher Beta and Lawnchair. They do not face this issue.

This issue doesn't seem to be tied to setting wallpaper. Once, I observed
this when returning to home screen after using other apps.

The common thing between these two scenarios was the particular animation.

**Comment by ecarl65** [comment]
I have a Pixel 7 and the latest Niagara beta and I have this issue often. Except it happens in many contexts, not just setting the wallpaper. Many times merely going to the Niagara home screen it occurs.

**Comment by EtzBetz** [comment]
I have this issue often as well, but with the Spotify view when switching to the Homescreen:

https://www.reddit.com/r/NiagaraLauncher/comments/10hspcj/_/

**Comment by mjdjohansson** [comment]
Same here. Also on a Pixel 7 with Android 13.

**Comment by ecarl65** [comment]
I *think* I just discovered a little more about this bug. I just changed the animation settings in developer mode from 0.5 back to the default of 1 and the bug seems to have disappeared.

**Comment by EtzBetz** [comment]
> I *think* I just discovered a little more about this bug. I just changed the animation settings in developer mode from 0.5 back to the default of 1 and the bug seems to have disappeared. 

I have and always had the animation settings at 1, and the bug is still occurring for me. Maybe it's occurring more often with faster animation speed, tho?

**Comment by ecarl65** [comment]
Ah, okay. Good to know. Thanks. I guess maybe it's happening less often so I haven't observed it yet.

**Comment by ecarl65** [comment]
I just experienced it at 1x animation speed, sorry for the false claim. It does seem to be less frequent though, in case that helps in debugging.

**Comment by Amorphous404** [comment]
Not sure if the other model users are facing the same bug since the article only mentions Pixel 7 Pro. However, it could be related, since it appears only on Pixel devices.

https://www.xda-developers.com/android-13-qpr2-beta-32-display-fix/

**Comment by Amorphous404** [comment]
Display & Graphics

 Fix for issue occasionally causing display flicker or artifacts in certain apps or conditions *[1]

[Google Pixel Update March 2023](https://support.google.com/pixelphone/thread/205950390/google-pixel-update-march-2023?hl=en)

**Comment by EtzBetz** [comment]
> Display & Graphics
> 
>  Fix for issue occasionally causing display flicker or artifacts in certain apps or conditions *[1]
> 
> [Google Pixel Update March 2023](https://support.google.com/pixelphone/thread/205950390/google-pixel-update-march-2023?hl=en)
> 
> 

I've got the new update now on my P6 and the issue still persists, but the area visually got bigger. I'm going to try to capture it..

**Comment by ecarl65** [comment]
I agree, that's exactly my experience. I still have the issue and the area
of the artifact is bigger.

**Comment by RDHose** [comment]
I have a Pixel 6a with the March Update and I too am having this issue when returning to the home screen randomly (not related to the wallpaper picker).  I have not figured out a way to duplicate it, as it happens completely at random.  If I try to duplicate it right after it never happens.  Seems like after the device is asleep for a bit and I go into a app like messages then go back to my home screen it happens.  I believe I started to notice this with the February update.  Also it should be noted I use Muzei as my live wallpaper background that pulls images from a local cache from downloaded photos that I imported from my Google Photos.

**Comment by RiccardoMargiotta** [comment]
Same for me on Pixel 6. I used to notice it more when I was using Muzei to rotate wallpapers, it happens less often since I switched to a static wallpaper.

**Comment by seankhl** [comment]
I face this bug too. It's the biggest bug I face with Niagara, and I do not think it should be considered low priority. Much appreciation for the project in general and I am not suggesting I am entitled to a fix or prioritization of this issue, just sharing my opinion.

**Comment by tomdod** [comment]
I also have this issue on Pixel 8 running Android 14. I'm unable to identify an app which consistently reproduces this, but I get it a few times a day.

**Comment by RobertHarnett** [comment]
Well I guess I clearly am the biggest loser... It's very reproducible for me, it happens for every. Single. App.

Using Pixel 7 Pro in the latest Android 14 Beta 
But it also occured before I joined the beta.

It also happened on my Pixel 6 Pro when I previously had that.

https://github.com/8bitPit/Niagara-Issues/assets/18248672/4d0697f7-5dd8-4395-8d7a-c9f3d3118e8c

**Comment by maceip** [comment]
Niagara pro Google pixel 8 user , using background from default pixel 

When switching to home view, the ui flickers / glitches very regularly

My firefox search widget also disappears regularly

Setup:

![Screenshot_20240108-104848](https://github.com/8bitPit/Niagara-Issues/assets/804368/4fb4008a-6828-48c7-9ad6-f1b39d432342)

**Comment by Stono** [comment]
Adding my +1 here pixel 8 android 14 all up to date.
Such a shame as I just discovered this launcher and on the whole love it!

**Comment by TFackler** [comment]
Issue also appeared on:
Pixel 8 Pro
Android 14, Build AP1A.240405.002.B1
Niagara Launcher Version 1.11.5

**Comment by RobertHarnett** [comment]
Now I don't wanna build anyone's hopes up, but I'm currently running the Android 15 Beta 4 on my Pixel 7 Pro, and as far as I can tell this issue seems to be fixed! 🎉

I've not experienced it once, not even the small lines I got around an icon every time I closed an app. I will be sure to come back and give an update if I see anything but until then... WOOOO!!!!

**Comment by netguino** [comment]
Pixel 8, non-pro. since the beta update, things are better, but still happen, but extremely occasionally now, as opposed to every 4th unlock/return to home. 

I have had about 3 instances of this happen, 1 the standard glitched homescreen, and 2 instead of the home screen list, I've been "stuck" in a non functional complete list of all apps ( like the app drawer )

**Comment by TFackler** [comment]
We have officially reported this issue to Google and hope they will fix it.

Please upvote it (the `+1` in the top right corner) to get the Android development team to see it hopefully :)
[Android Issue Tracker - Yellow/Green lines artifacts...](https://issuetracker.google.com/issues/368256328)

**Comment by ZtereoHYPE** [comment]
@TFackler Thank you for the official report!

Unfortunately the issue has been marked as Won't Fix by google because debugging information was not provided. Should another one be open perhaps?
